ui/components/component-library/avatar-with-badge/README.mdx

```diff
@@ -0,0 +1,33 @@
+import { Story, Canvas, ArgsTable } from '@storybook/addon-docs';
+
+import { AvatarWithBadge } from './avatar-with-badge';
+
+# AvatarWithBadge
+
+The `AvatarWithBadge` is a wrapper component that adds badge display options to avatars.
+
+<Canvas>
+  <Story id="ui-components-component-library-avatar-with-badge-avatar-with-badge-stories-js--default-story" />
+</Canvas>
+
+## Props
+
+The `AvatarWithBadge` accepts all props below 
+
+<ArgsTable of={AvatarWithBadge} />
+
+### badgePosition
+
+Use the `badgePosition` prop to set the position of the badge, it can have two values `Top` and `Bottom`
+
+### badge
+
+Used to define the badge component to be rendered inside the `AvatarWithBadge`
+
+### badgeProps
+
+The required props to be passed to the badge.
+
+### children
```

Comment:
Apologies, because I think I probably suggested this convention but could we match how storybook displays the story names in title case?

```suggestion
### Badge Position

Use the `badgePosition` prop to set the position of the badge, it can have two values `Top` and `Bottom`

### Badge

Used to define the badge component to be rendered inside the `AvatarWithBadge`

### Badge Props

The required props to be passed to the badge.

### Children
```